### ISSUE TITLE
Feature request: Axis label spaning only a subset of subplots

### ISSUE THREAD

**Issue by Kexikus** [lukelbd/proplot#352]
This is a feature request as I have been unable to find a working solution right now. In short, I would like to see the option of label spaning for only some rows/columns.
For example, I have a plot with three rows and I would like the y label to span the first two rows (as they have the same y axis) but keep a different/unique y label for the last row.

And the same would go for x labels as well of course.

**Comment by lukelbd** [comment]
Makes sense, esp. in the context of #200. We should be able to both restrict *spanning* labels and *shared* labels to specific row/column groups with a few `subplots()` keyword arguments.
